json/src/main/java/com/esri/json/hadoop/UnenclosedJsonRecordReader.java

```diff
@@ -213,33 +216,42 @@ public boolean next(LongWritable key, Text value) throws IOException {
 		 * We will count '{' and '}' to find the beginning and end of each record, while ignoring braces in string literals
 		 */
 		
-		if (readerPosition >= end) {
-			return false;
-		}
-		
 		int chr = 0;
 		int brace_depth = 0;
 		char lit_char = 0;
 		boolean first_brace_found = false;
+
+		// The case of split point exactly at whitespace between records,
+		// is handled by forcing the record following to the split following,
+		// in the interest of better balancing the splits, by consuming the
+		// whitespace before checking the end of the split.
+		if (!firstBraceConsumed) {  // That should only ever be true on the very first read in the split
```

Comment:
For whitespace between records, eat whitespace before checking end of split.  Tested by `TestWhitespace` with `unenclosed-json-return.json`.